lib/api/user_token_service.rb

```diff
@@ -1,23 +1,28 @@
 module Api
   class UserTokenService
+    TYPES = %w(api ui ws).freeze
+    # Additional Requester type token ttl's for authentication
+    TYPE_TO_TTL_OVERRIDE = {'ui' => ::Settings.session.timeout}.freeze
+
     def initialize(config = Settings, args = {})
       @config = config
       @svc_options = args
-      @token_mgr = new_token_mgr(base_config[:module], base_config[:name], api_config)
     end
 
-    attr_accessor :token_mgr
-
-    # Additional Requester type token ttl's for authentication
-    #
-    REQUESTER_TTL_CONFIG = {"ui" => :ui_token_ttl}.freeze
+    def token_mgr(type)
+      @token_mgr ||= {}
+      case type
+      when 'api', 'ui' # The default API token and UI token share the same TokenStore
+        @token_mgr['api'] ||= new_token_mgr(base_config[:module], base_config[:name], api_config)
+      when 'ws'
+        @token_mgr['ws'] ||= TokenManager.new('ws', :token_ttl => ::Settings.session.timeout)
```

Comment:
Why there was a 5 minutes before?

In previous discussions we had an idea to make the token short-lived, but then we went with totally separated tokens. The separated tokens are way better security mechanism than time-elapsing tokens.

The risks of short-lived/non-separated tokens are high. The token for WS needs to be stored in a cookie by client. Since browsers do not apply same-origin cookie policy on WS, the cookie can be hijacked. Hence, we want to have WS token separate. Then, we can mitigate cross-site WS hijacking by having the server validate 'Origin' header. So, now we have other mechanisms (separate token) to secure it and we don't need the band-aid of having the tokens short-lived.